index.ts

```diff
@@ -1,37 +1,37 @@
 import * as pulumi from "@pulumi/pulumi";
 import * as eks from "@pulumi/eks";
 
+// Get the current stack name
+const currentStackName = pulumi.getStack();
+
 // Grab some configuration values
 const config = new pulumi.Config();
 const minClusterSize = config.getNumber("minClusterSize") || 3;
 const maxClusterSize = config.getNumber("maxClusterSize") || 6;
 const desiredClusterSize = config.getNumber("desiredClusterSize") || 3;
 const eksNodeInstanceType = config.get("eksNodeInstanceType") || "t3.medium";
-const infraOrgName = config.require("infraOrgName");
-const infraProjName = config.require("infraProjName");
-const infraStackName = config.require("infraStackName");
+const baseOrgName = config.get("baseOrgName") || "zephyr";
+const baseProjName = config.get("baseProjName") || "zephyr-infra";
+const baseStackName = config.get("baseStackName") || currentStackName;
 
 // Create a StackReference to get Kubeconfig from base stack
-const infraSr = new pulumi.StackReference(`${infraOrgName}/${infraProjName}/${infraStackName}`);
-const infraVpcId = infraSr.getOutput("vpcId");
-const infraPrivSubnetIds = infraSr.getOutput("privSubnetIds");
-const infraPubSubnetIds = infraSr.getOutput("pubSubnetIds");
+const baseSr = new pulumi.StackReference(`${baseOrgName}/${baseProjName}/${baseStackName}`);
```

Comment:
Assuming we deploy the different layered stacks in the same organization, and keep the stack name consistent for stacks making up the same environment (e.g. `dev`), wouldn't it be better to showcase the `getOrganization` and `getStack` calls from our core SDK?

https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/#getOrganization
https://www.pulumi.com/docs/reference/pkg/nodejs/pulumi/pulumi/#getStack

Reply:
Overlooked you already use `getStack`. Remark still stands for `getOrganization` though.

Reply:
Resolved in bef672c. I added `getOrganization` to provide a default organization name if the user does not supply an organization name in the stack configuration. Great idea!